palace/models/postoperator.cpp

```diff
@@ -409,7 +419,71 @@ void ScaleGridFunctions(double L, int dim, bool imag, T &E, T &B, T &V, T &A)
   }
 }
 
-}  // namespace
+template <typename T>
+void DimensionalizeGridFunctions(Units &units, bool imag, T &E, T &B, T &V, T &A)
+{
+  if (E)
+  {
+    units.DimensionalizeInPlace<Units::ValueType::FIELD_E>(*E);
+    units.DimensionalizeInPlace<Units::ValueType::FIELD_E>(E->Real().FaceNbrData());
+    if (imag)
+    {
+      units.DimensionalizeInPlace<Units::ValueType::FIELD_E>(E->Imag().FaceNbrData());
+    }
+  }
+  if (B)
+  {
+    units.DimensionalizeInPlace<Units::ValueType::FIELD_B>(*B);
+    units.DimensionalizeInPlace<Units::ValueType::FIELD_B>(B->Real().FaceNbrData());
+    if (imag)
+    {
+      units.DimensionalizeInPlace<Units::ValueType::FIELD_B>(B->Imag().FaceNbrData());
+    }
+  }
```

Comment:
I tried this change (so that we don't need to pass `imag`):
```
  if (E)
  {
    units.DimensionalizeInPlace<Units::ValueType::FIELD_E>(*E);
    units.DimensionalizeInPlace<Units::ValueType::FIELD_E>(E->Real().FaceNbrData());
    if (E->Imag())
    {
      std::cout << "I AM CALLED" << std::endl;
      units.DimensionalizeInPlace<Units::ValueType::FIELD_E>(E->Imag().FaceNbrData());
    }
  }
```
and run it with the spheres example. The code run and "I AM CALLED" was not printed (which I added to check that `E->Imag()` doesn't always trivially evaluate to true).

Reply:
I left a comment about templates below, but if we make the assumption that the arguments are palace gridfunctions (which we implicitly do in how we access Real and Imag), we can be explicit and call `HasImag()`)

Reply:
We can also refactor `ScaleGridFunctions` to not take `imag`